apps/dashboard/app/controllers/settings_controller.rb

```diff
@@ -0,0 +1,29 @@
+class SettingsController < ApplicationController
+
+  ALLOWED_SETTINGS = [:profile]
+
+  def update
+    new_settings = read_settings(settings_param)
+    CurrentUser.update_user_settings(new_settings) unless new_settings.empty?
+
+    logger.info "settings: updated user settings to: #{new_settings}"
+    respond_to do |format|
+      format.html { redirect_to root_url, notice: I18n.t('dashboard.settings_updated') }
+      format.json { head :no_content }
+    end
+  end
+
+  private
+
+  def settings_param
+    params.require(:settings).permit(ALLOWED_SETTINGS) if params[:settings].present?
+  end
+
+  def read_settings(params)
+    settings = {}
+    params.each do |key, value|
+      settings[key] = value
+    end if params
+    return settings
```

Comment:
```suggestion
    {}.tap |settings|
      params.each do |key, value|
        settings[key] = value
      end
    end
```
If you find yourself initializing a return value, then iterating, then returning, there's likely a more Ruby-esqe way to do the same.